From 2a1e4723533d1021df87cc973e9736a5def6b33b Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex@uq.edu.au>
Date: Wed, 21 Aug 2019 13:54:56 +1000
Subject: [PATCH] joyent/node-triton#265 Un-hide --act-as option and add
 environment var Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by:
 Trent Mick <trent.mick@joyent.com> Approved by: Trent Mick
 <trent.mick@joyent.com>

---
 lib/cli.js    | 10 +++++-----
 lib/config.js |  4 ++++
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/lib/cli.js b/lib/cli.js
index ca6acd0..43ca473 100644
--- a/lib/cli.js
+++ b/lib/cli.js
@@ -88,11 +88,11 @@ var OPTIONS = [
     {
         names: ['act-as'],
         type: 'string',
-        help: 'Masquerade as the given account login name. This can only ' +
-            'succeed for operator accounts. Note that accesses like these ' +
-            'audited on the CloudAPI server side.',
-        helpArg: 'ACCOUNT',
-        hidden: true
+        help: 'Masquerade as the given account login name. This is useful ' +
+            'for operator accounts and members of cross-account roles. Note ' +
+            'that accesses like these are audited on the CloudAPI server ' +
+            'side. Environment: TRITON_ACT_AS=ACCOUNT.',
+        helpArg: 'ACCOUNT'
     },
     {
         names: ['user', 'u'],
diff --git a/lib/config.js b/lib/config.js
index b5708b2..ee742b8 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -283,6 +283,10 @@ function _loadEnvProfile(profileOverrides) {
     }
     envProfile.url = process.env.TRITON_URL || process.env.SDC_URL;
     envProfile.keyId = process.env.TRITON_KEY_ID || process.env.SDC_KEY_ID;
+    var actAs = process.env.TRITON_ACT_AS;
+    if (actAs) {
+        envProfile.actAsAccount = actAs;
+    }
 
     if (process.env.TRITON_TLS_INSECURE) {
         envProfile.insecure = common.boolFromString(
-- 
2.21.0

